src/components/util/patronBlocks.test.js

```diff
@@ -0,0 +1,53 @@
+import getRenewalPatronBlocksFromPatronBlocks from './patronBlocks';
+
+const patronBlocks = [{
+  borrowing: true,
+  desc: 'Sample',
+  id: 'f1e0d3e2-fa48-4a82-b371-bea4e44178ab',
+  patronMessage: '',
+  renewals: true,
+  requests: true,
+  staffInformation: '',
+  type: 'Manual',
+  userId: 'e6dc87a3-591b-43e0-a768-d3552b44878b',
+  metadata: { createdDate: '2022-03-01T03:22:48.262+00:00', updatedDate: '2022-03-01T03:22:48.262+00:00' }
+},
+{
+  borrowing: false,
+  blockRenewals: true,
+  desc: 'Test2',
+  id: 'f1e0d3e2-fa48-4a82-b371-bea4e44178ab',
+  patronMessage: '',
+  renewals: false,
+  requests: false,
+  staffInformation: '',
+  type: 'Manual',
+  userId: 'e6dc87a3-591b-43e0-a768-d3552b44878b',
+  metadata: { createdDate: '2022-03-01T03:22:48.262+00:00', updatedDate: '2022-03-01T03:22:48.262+00:00' }
+}];
+
+
+const patronBlocks2 = [{
+  borrowing: true,
+  desc: 'Sample',
+  id: 'f1e0d3e2-fa48-4a82-b371-bea4e44178ab',
+  patronMessage: '',
+  renewals: false,
+  requests: true,
+  staffInformation: '',
+  type: 'Manual',
+  userId: 'e6dc87a3-591b-43e0-a768-d3552b44878b',
+  metadata: { createdDate: '2022-03-01T03:22:48.262+00:00', updatedDate: '2022-03-01T03:22:48.262+00:00' }
+}];
+
+
+describe('patronBlocks Filter component', () => {
+  it('if it filters', () => {
+    const data = getRenewalPatronBlocksFromPatronBlocks(patronBlocks);
+    expect(data).toStrictEqual(patronBlocks);
+  });
+  it('if it filters renewal blocked data', () => {
+    const data = getRenewalPatronBlocksFromPatronBlocks(patronBlocks2);
+    expect(data).toStrictEqual([]);
+  });
```

Comment:
Based on the second test, I would have expected `data` in the first to exclude `Test2`. If I am mistaken, can you add a comment explaining why this is the correct behavior? And if I'm not, then I think you found a bug, and if you agree we should file a new UIU ticket.

Reply:
It is checking for the blockedrenewal scenario so it returns both data.

Reply:
Ah, OK. It wasn't immediately obvious from the tests that `getRenewalPatronBlocksFromPatronBlocks` checks the truthiness of `renewals` _or_ `blockRenewals`. Man, we really hurt for documentation about our business logic :(